app/controllers/requests_controller.rb

```diff
@@ -0,0 +1,52 @@
+class RequestsController < ApplicationController
+  before_action :require_login
+
+  def index
+    @requests = Request.all
+  end
+
+  def new
+    @request = Request.new
+  end
+
+  def create
+    @request = Request.new(request_attributes)
+    @request.requester = current_user
+    if @request.save
+      redirect_to request(@request)
+    else
+      flash.new[:alert] = @request.errors.full_messages.join(", ")
+      render 'new'
+    end
+  end
+
+  def show
+    @request = Request.find_by(params[:id])
```

Comment:
Does this work? We might have to use `@request = Request.find_by(id: params[:id])`.